src/twisted/python/failure.py

```diff
@@ -516,7 +516,7 @@ def throwExceptionIntoGenerator(self, g):
         """
         # Note that the actual magic to find the traceback information
         # is done in _findFailure.
-        return g.throw(self.type, self.value, self.tb)
+        return g.throw(self.value.with_traceback(self.tb))
```

Comment:
We were already using this code in CI with Python 3.12

I guess that the warning was raised, but there was no check for it.

It would be nice to have an explicit test that no warnings are raised, on any Python version.